volatility3/framework/layers/intel.py

```diff
@@ -321,53 +378,48 @@ def _page_is_valid(entry: int) -> bool:
-                    if swap_layer_name:
-                        return swap_offset, 1 << excp.invalid_bits, swap_layer_name
-                raise exceptions.SwappedInvalidAddressException(layer_name = excp.layer_name,
-                                                                invalid_address = excp.invalid_address,
-                                                                invalid_bits = excp.invalid_bits,
-                                                                entry = excp.entry,
-                                                                swap_offset = swap_offset)
-            raise
+    def _handle_page_fault(self, layer_name, offset, invalid_bits, entry, description):
+        kernel = self._kernel
+        if kernel is None:
+            raise exceptions.PagedInvalidAddressException(self.name, offset, invalid_bits, entry, "kernel module not found!")
+
+        tbit = bool(entry & (1 << 11))
+        pbit = bool(entry & (1 << 10))
+        vbit = bool(entry & 1)
+        entry ^= self._get_invalid_pte_mask(kernel)
+
+        # Handle Swap failure
+        if (not tbit and not pbit and not vbit) and ((entry >> self._swap_bit_offset) != 0):
+            swap_offset = entry >> self._swap_bit_offset << invalid_bits
+            n = (entry >> self._get_PageFileLow_shift(kernel)) & 0xF
+
+            if self.config.get('swap_layers', False):
+                swap_layer_name = self.config.get(
+                    interfaces.configuration.path_join('swap_layers', 'swap_layers' + str(n)), None)
+                if swap_layer_name:
+                    return swap_offset, 1 << invalid_bits, swap_layer_name
+            raise exceptions.SwappedInvalidAddressException(layer_name = layer_name,
+                                                            invalid_address = offset,
+                                                            invalid_bits = invalid_bits,
+                                                            entry = entry,
+                                                            swap_offset = swap_offset)
+        raise super()._handle_page_fault(layer_name, offset, invalid_bits, entry, description)
 
 
 ### These must be full separate classes so that JSON configs re-create them properly
 
 
 class WindowsIntel(WindowsMixin, Intel):
-
-    def _translate(self, offset):
-        return self._translate_swap(self, offset, self._page_size_in_bits)
+    pass
 
 
 class WindowsIntelPAE(WindowsMixin, IntelPAE):
-
-    def _translate(self, offset: int) -> Tuple[int, int, str]:
-        return self._translate_swap(self, offset, self._bits_per_register)
+    pass
 
 
 class WindowsIntel32e(WindowsMixin, Intel32e):
     # TODO: Fix appropriately in a future release.
-    # Currently just a temporary workaround to deal with custom bit flag
+    # Currently just a temprorary workaround to deal with custom bit flag
```

Comment:
It seems to right, before updated version.

```suggestion
    # Currently just a temporary workaround to deal with custom bit flag
```

Reply:
@paulkermann I think this has been resolved not yet :)

Reply:
Ah woops